src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/TypeExtension.cs

```diff
@@ -44,89 +39,78 @@ public TypeExtension(string typeName)
         public TypeExtension(Type type)
         {
             _type = type ?? throw new ArgumentNullException(nameof(type));
-            // we would like to set TypeName here, but we can't because we can't resolve its namespace
         }
 
         /// <summary>
-        ///  Return an object that should be set on the targetObject's targetProperty
-        ///  for this markup extension.  TypeExtension resolves a string into a Type
-        ///  and returns it.
+        /// Return an object that should be set on the targetObject's targetProperty
+        /// for this markup extension. TypeExtension resolves a string into a Type
+        /// and returns it.
         /// </summary>
         /// <param name="serviceProvider">Object that can provide services for the markup extension.</param>
-        /// <returns>
-        ///  The object to set on this property.
-        /// </returns>
+        /// <returns>The object to set on this property.</returns>
         public override object ProvideValue(IServiceProvider serviceProvider)
         {
             // If a type was supplied, no context nor type name are needed
-
             if (_type != null)
             {
                 return _type;
             }
 
-            // Validate the initialization
-            
+            // Validate the initialization.
             if (_typeName == null)
             {
                 throw new InvalidOperationException(SR.MarkupExtensionTypeName);
             }
 
             // Get the IXamlTypeResolver from the service provider
-
             ArgumentNullException.ThrowIfNull(serviceProvider);            
 
             IXamlTypeResolver xamlTypeResolver = serviceProvider.GetService(typeof(IXamlTypeResolver)) as IXamlTypeResolver;
-            if( xamlTypeResolver == null )
+            if( xamlTypeResolver == null)
             {
-                throw new InvalidOperationException( SR.Format(SR.MarkupExtensionNoContext, GetType().Name, "IXamlTypeResolver" ));
+                throw new InvalidOperationException(SR.Format(SR.MarkupExtensionNoContext, GetType().Name, nameof(IXamlTypeResolver)));
             }
 
             // Get the type
-            
             _type = xamlTypeResolver.Resolve(_typeName);
-
             if (_type == null)
             {
                 throw new InvalidOperationException(SR.Format(SR.MarkupExtensionTypeNameBad, _typeName));
             }
-            
+
             return _type;
-            // we could cash the result of type as _type, but it might cause some problems, and in theory we shouldn't need to
```

Comment:
I think we should leave this comment here, it provides information regarding caching of _type.

Reply:
Removed this comment as we actually do cache the value - see L54 above: https://github.com/dotnet/wpf/pull/207/files/64307878bc044c6bfc8dcc6e00e2f3668dba7511#diff-b9a01bc218eb41c6b22457d0b6aedb948dbffe01779ef042058a7ba7a726fdf6R54